main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala

```diff
@@ -228,17 +228,17 @@ object EntryPoint {
 
     // NB: Getting the type directly from the scheme assumes the function is not polymorphic.
     // This is a valid assumption with the limitations we set on the entry point.
-    val func = TypedAst.Expr.Def(oldEntryPoint.sym, oldEntryPoint.spec.declaredScheme.base, SourceLocation.Unknown)
-
+    val itpe = oldEntryPoint.spec.declaredScheme.base
+    val defSym = Ast.DefSymUse(oldEntryPoint.sym, SourceLocation.Unknown)
+    val args = List(TypedAst.Expr.Cst(Constant.Unit, Type.Unit, SourceLocation.Unknown))
     // func()
-    val call = TypedAst.Expr.Apply(func, List(TypedAst.Expr.Cst(Constant.Unit, Type.Unit, SourceLocation.Unknown)), oldEntryPoint.spec.declaredScheme.base.arrowResultType, oldEntryPoint.spec.declaredScheme.base.arrowEffectType, SourceLocation.Unknown)
+    val call = TypedAst.Expr.ApplyDef(defSym, args, itpe, itpe.arrowResultType, itpe.arrowEffectType, SourceLocation.Unknown)
```

Comment:
defSym should be symUse here right?

Reply:
Just so we are on the same page- in what phase do we drop DefSymUse and just use DefnSym? Is it from SimplifiedAst and forward?

Reply:
Yes from `SimplifiedAst` and later it is just a `DefnSym`